lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoader.java

```diff
@@ -165,27 +231,35 @@ public void onNext(final AllocatedEvaluator allocatedEvaluator) {
       }
 
       final int evaluatorsForComputeRequest = numComputeRequestsToSubmit.decrementAndGet();
-      LOG.log(Level.FINE, "Evaluators for compute request: {0}", evaluatorsForComputeRequest);
 
       if (evaluatorsForComputeRequest >= 0) {
```

Comment:
because you need to release the gate when you are doing allocating the compute requests. After those, then the data requests allocations proceed. Is a weird thing how they use the countDownLatch, but didn't want to change that.